word_guess.rb

```diff
@@ -0,0 +1,182 @@
+
+  def intro
+    puts "Welcome to our game! Try to solve the word by guessing letters.
+    Once sleepy cat wakes up (loses Zs), the game is over!\n\n"
+    "word: #{underscores}\n"
+  end
+
+  def cat_message
+    cat_message = ["\nWrong letter.. Don't wake sleep cat!\n\n",
+      "\nLetter does not exist! Sleepy cat is less sleepy.\n\n",
+      "\nGuess again! Sleepy cat is stirring.\n\n",
+      "\nNot even close. Sleepy cat is about to wake up.\n\n",
+      "\nNo more guesses. Sleepy cat has decided to bike away.\n\n"]
+  end
+
+  def sleepy_cat
+    sleepy_cat = [
+    "     Z",
+    "       Z",
+    "         Z",
+    "           Z",
+    "             Z",
+    "                  |\\     _,,,---,,_
+                   /,`.-'`'    -.  ;-;;,_
+                  |,4-  ) )-,_..;\\  ( `'-'
+                 '---''(_/--'  `-'\\_)
+                 " ]
+
+  end
+
+  def bike_cat
+    bike_cat = ['
+                    ________________
+                   |      GAME      |_____    __
+                   |     OVER!!!    |     |__|  |_________
+                   |________________|     |::|  |        /
+      /\**/\       |                \.____|::|__|      <
+      ( o_o  )_    |                      \::/  \._______\
+      (u--u   \_)  |
+       (||___   )==\
+      ,dP"/b/=( /P"/b\
+      |8 || 8\=== || 8
+      `b,  ,P  `b,  ,P
+        """`     """`  ']
+  end
+
+  def new_letter
+    # prompt for very first letter guess
+    print "Enter a letter: "
+    @letter = gets.chomp
+    puts
+  end
+
+  def words_and_guessed
+    print "word: #{@underscores.join(' ')}\n"
+    print "guessed letters: #{@guessed_letters.join (' ')}\n\n"
+  end
+
+  def try_again
+    print "Please try again. Enter a letter: "
+    @letter = gets.chomp
+  end
+
+  def letter_check
+    # check to see if it's a letter
+    while true
+      if @letter[/[a-zA-Z]+/] == @letter && @letter.length == 1
+        # check to see if it's ONE @letter
+          if @guessed_letters.include? @letter
+            try_again
+            # break
+          else
+            # places guessed @letters into an array
+            @guessed_letters.push(@letter)
+            # puts @guessed_letters
+            # drop_z
+            break
+          end
+        else
+          try_again
+          # break
+        end
+    end
+  end
+
+  def underscores
+    @underscores = []
+    (0..(@answer.length-1)).each do
+       @underscores = @underscores.push("_ ")
+    end
+    @underscores.join(' ')
+  end
+
+  def replace_underscores
+    @answer.length.times do |x|
+      if @letter == @answer[x]
+        @underscores[x] = "#{@letter} "
+        #@valid_under[0][x] = "#{@letter}"
+      end
+    end
+  end
+
+  def drop_z
+    #take away a Z from the cat
+    unless @answer.include? @letter
+      # user enters a letter - > z
+      # automatically gets pushed into guessed_letters
+      if @guessed_letters[0..@guessed_letters.length-2].include? @letter
+      else
+        @num += 1
```

Comment:
I might recommend using a more meaningful variable name that describes _what_ this number is used for.